scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -1116,6 +1150,84 @@ public FilteredStatistics getFilteredStatistics(final String workflowName, Strin
+        long endTime = new Date().getTime();
+        long timeInterval = 0;
+
+        switch (timeWindow) {
+            case DAILY:
+                startTime = endTime - TimeUnit.HOURS.toMillis(24);
+                timeInterval = TimeUnit.HOURS.toMillis(1);
+                break;
+            case WEEKLY:
+                startTime = endTime - TimeUnit.DAYS.toMillis(7);
+                timeInterval = TimeUnit.DAYS.toMillis(1);
+                break;
+            case MONTHLY:
+                startTime = endTime - TimeUnit.DAYS.toMillis(30);
+                timeInterval = TimeUnit.DAYS.toMillis(1);
+                break;
+            case YEARLY:
+                Calendar cal = Calendar.getInstance();
+                cal.add(Calendar.YEAR, -1);
+                startTime = cal.getTimeInMillis();
+                timeInterval = TimeUnit.DAYS.toMillis(30);
+                break;
+        }
+
+        List<JobInfo> completedJobsWithIssues = getFilteredJobs(workflowName,
+                                                                user,
+                                                                tenant,
+                                                                startTime,
+                                                                endTime,
+                                                                Collections.singletonList(JobStatus.FINISHED),
+                                                                true);
+        List<JobInfo> completedJobsWithoutIssues = getFilteredJobs(workflowName,
+                                                                   user,
+                                                                   tenant,
+                                                                   startTime,
+                                                                   endTime,
+                                                                   Collections.singletonList(JobStatus.FINISHED),
+                                                                   false);
+        List<JobInfo> failedJobs = getFilteredJobs(workflowName,
+                                                   user,
+                                                   tenant,
+                                                   startTime,
+                                                   endTime,
+                                                   ImmutableSet.of(JobStatus.CANCELED,
+                                                                   JobStatus.FAILED,
+                                                                   JobStatus.KILLED),
+                                                   null);
+        completedJobsWithIssues.addAll(failedJobs);
+
+        Map<Integer, Integer> jobsWithoutIssuesCount = new TreeMap<>();
+        Map<Integer, Integer> jobsWithIssuesCount = new TreeMap<>();
+        int position = 0;
+        for (long i = startTime; i < endTime; i = i + timeInterval, position++) {
```

Comment:
I would have tried with joda .. but not sure about its benefits. So ok.

Reply:
In general, we should consider time zones. So I think this code should be improved to have an extra parameter containing the time zone.

Instead of using a dedicated library, we can use https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html